pkg/client/cache/index.go

```diff
@@ -24,9 +24,13 @@ func ImageStreamReferenceIndexFunc(obj interface{}) ([]string, error) {
 			}
 			from := trigger.ImageChange.From
 
-			// the default imagechange trigger has no "from" value, it
-			// uses the input image of the buildconfig as the imagestream
-			// to trigger off, so we need to look that up.
+			// We're indexing on the imagestream referenced by the imagechangetrigger,
+			// however buildconfigs allow one "default" imagechangetrigger in which
+			// the ICT does not explicitly indicate the image it is triggering on,
+			// instead it triggers on the image being used as the builder/base image
+			// as referenced in the build strategy, so if this is an ICT w/ no
+			// explicit image reference, use the image referenced by the strategy
+			// because this is the default ICT.
```

Comment:
As a note for the next version of the API, we should default this onto the ICT so that it's explicit rather than being implicit.  Implicit creates more complicated code and complicates clients.